include/utils/kahan.hpp

```diff
@@ -43,4 +45,94 @@ T kahan(InputIt first, InputIt last, T init) {
     return sum;
 }
 
+template<class InputIt, class T>
+T kahanBabushka(InputIt first, InputIt last, T init) {
+    T sum = init;
+    T cs = 0.;
+    T ccs = 0.;
+    T c = 0.;
+    T cc = 0.;
+
+    for (auto input = first; input != last; input++) {
+        T t = sum + *input;
```

Comment:
We should create a variable called `auto input_i = *input` at the head of the loop, and then you can replace all instances of `*input` with `input_i`. That way, we are sure that this is only dereferenced once.